src/domain/relay/contracts/safe-contract.helper.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@nestjs/common';
+import { getSafeSingletonDeployment } from '@safe-global/safe-deployments';
+import { Abi, isHex } from 'viem';
+import { Hex } from 'viem/src/types/misc';
+import { ContractFunctionArgsMapper } from '@/domain/relay/entities/contract-function.entity';
+import { ContractHelper } from '@/domain/relay/contracts/contract.helper';
+
+@Injectable()
+export class SafeContractHelper extends ContractHelper {
+  protected readonly abi: readonly unknown[] | Abi;
+
+  private static SUPPORTED_SAFE_VERSION = '1.3.0';
+
+  static EXEC_TRANSACTION: ContractFunctionArgsMapper<{
+    data: Hex;
+    to: Hex;
+    value: bigint;
+  }> = {
+    name: 'execTransaction',
+    mapper(args: readonly unknown[]): { data: Hex; to: Hex; value: bigint } {
+      if (!isHex(args[2]) || !isHex(args[0]) || typeof args[1] !== 'bigint')
+        throw Error('Unexpected type for execTransaction args');
+      return { data: args[2], to: args[0], value: args[1] };
+    },
+  };
```

Comment:
I know it's likely a TODO, but this should support every contract method for the relayer.

Reply:
From the current implementation of the Safe Gelato Relay, we decode the `execTransaction` call but not the others. Support for the other functions is limited to checking if it's a call to the Safe Contract (from the `data` of the `execTransaction` call). This is already part of the PR: https://github.com/safe-global/safe-client-gateway/pull/680/files#diff-0fbe9bd32fe04ec35fc1fab18a5bf2db84a4c21877551de1311be99dc07b258fR72

Reply:
I see, oversight on my behalf - apologies. Will resolve this!